packages/yarnpkg-core/sources/StreamReport.ts

```diff
@@ -607,8 +618,8 @@ export class StreamReport extends Report {
       let progressBar = ``;
 
       if (typeof progress.lastScaledSize !== `undefined`) {
-        const ok = this.progressStyle.chars[0].repeat(progress.lastScaledSize);
-        const ko = this.progressStyle.chars[1].repeat(this.progressMaxScaledSize - progress.lastScaledSize);
+        const ok = this.progressStyle!.chars[0].repeat(progress.lastScaledSize);
+        const ko = this.progressStyle!.chars[1].repeat(this.progressMaxScaledSize! - progress.lastScaledSize);
```

Comment:
Won't need some of those `!` if we use `progressStyle` to know whether progress is enabled or not
```suggestion
        const ok = this.progressStyle.chars[0].repeat(progress.lastScaledSize);
        const ko = this.progressStyle.chars[1].repeat(this.progressMaxScaledSize! - progress.lastScaledSize);
```